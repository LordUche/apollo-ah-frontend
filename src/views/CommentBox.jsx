import React from 'react';
import { PropTypes } from 'prop-types';
import LikeComment from '../components/LikeComment';


/**
 * Creates the JSX element used to display a single comment
 * @param {object} props The component props
 * date - Short format representation of the date the comment was made
 * body - The comment body
 * authorImage - The profile image of the  user
 * authorName - The firstname or username of the user
 * fullDate - Long format representation of the date the comment was made
 * @returns {Node} The generated JSX element
 */
export default function CommentBox({
  date,
  body,
  authorImage,
  authorName,
  fullDate,
  id,
  articleSlug,
  // handleLikes,
  // handleDislikes,
  likeCount,
  dislikeCount
}) {
  return (
    <div className="article-author">
      <img src={authorImage} alt={authorName} />
      <div className="comment-info">
        <h6>{authorName}</h6>
        <span title={fullDate}>{date}</span>
        <p>{body}</p>
      </div>
      <LikeComment
        id={id}
        slug={articleSlug}
        likeCount={likeCount}
        dislikeCount={dislikeCount}
      />
    </div>

  );
}

/** Prop types declaration */
CommentBox.propTypes = {
  authorName: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  authorImage: PropTypes.string.isRequired,
  fullDate: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  articleSlug: PropTypes.string.isRequired,
  // handleLikes: PropTypes.func.isRequired,
  // handleDislikes: PropTypes.func.isRequired,
  likeCount: PropTypes.number.isRequired,
  dislikeCount: PropTypes.number.isRequired,
};
