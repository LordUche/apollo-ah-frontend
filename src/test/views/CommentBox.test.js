import React from 'react';
import { shallow } from 'enzyme';
import CommentBox from '../../views/CommentBox';


describe('<CommentBox>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<CommentBox
      authorName="john"
      authorImage="image.jpg"
       body="text"
       date="2hrs ago"
       fullDate="July 4th 2019, 4:45pm"
        />);
    expect(wrapper).toMatchSnapshot();
  });
});
