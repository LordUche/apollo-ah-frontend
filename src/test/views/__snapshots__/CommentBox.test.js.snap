// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentBox> should render without crashing 1`] = `
<div
  className="article-author"
>
  <img
    alt="john"
    src="image.jpg"
  />
  <div
    className="comment-info"
  >
    <h6>
      john
    </h6>
    <span
      title="July 4th 2019, 4:45pm"
    >
      2hrs ago
    </span>
    <p>
      text
    </p>
  </div>
  <Connect(LikeComment) />
</div>
`;
